meter.go

```diff
@@ -106,33 +104,25 @@ func getCounter(m *Metrics, counterName string) Counter {
 	return c
 }
 
-// GetJSON filters counters by glob pattern and returns them as a json marshaled
-// map.
-func (m *Metrics) GetJSON(pattern string) ([]byte, error) {
-	return getJSON(m, pattern)
+// GetJSON filters counters by given predicate and returns them as a json
+// marshaled map.
+func (m *Metrics) GetJSON(predicate func(string) bool) []byte {
+	return getJSON(m, predicate)
 }
 
-func getJSON(m *Metrics, pattern string) ([]byte, error) {
-	g, err := glob.Compile(pattern)
-	if err != nil {
-		return nil, err
-	}
-
-	result := make(map[string]int64)
+func getJSON(m *Metrics, predicate func(string) bool) []byte {
+	result := make(map[string]Counter)
+	formatter := jsonFormatter{}
```

Comment:
Moving locking/unlocking from the top to some random place can cause a simple forgetting to lock or unlock. To avoid such cases and to reduce a WTF level let's use the following:

```go

func (f *foo) bar() {
f.mu.Lock()
defer f.mu.Unlock()

// other stuff
}
```